src/vmm/src/resources.rs

```diff
@@ -856,18 +856,15 @@ mod tests {
             rootfs_file.as_path().to_str().unwrap()
         );
 
-        match VmResources::from_json(
-            json.as_str(),
-            &default_instance_info,
-            HTTP_MAX_PAYLOAD_SIZE,
-            None,
-        ) {
-            Err(ResourcesError::Logger(LoggerConfigError::InitializationFailure { .. })) => (),
-            _ => unreachable!(),
-        }
-
-        // The previous call enables the logger. We need to disable it.
-        LOGGER.set_max_level(LevelFilter::Off);
+        assert!(matches!(
+            VmResources::from_json(
+                json.as_str(),
+                &default_instance_info,
+                HTTP_MAX_PAYLOAD_SIZE,
+                None,
+            ),
+            Err(Error::Logger(LoggerConfigError::File(_))),
+        ));
```

Comment:
Previously the logger could only be configured once, after which the API would return errors if you tried to re-intialize it. Now the logger can be configured, we don't need to disable it here.

Reply:
This comment was wrong.
```rust
LOGGER.set_max_level(LevelFilter::Off);
```
doesn't disable the logger (it doesn't close file descriptors to its write targets or drop itself) it only changes the filter log level.

I think previously it would first check the log level before attempting to take action, thus would hit the logger file error (otherwise it would hit the re-intialization error).

SInce the logger now allows re-configuration this error case doesn't happen so it hits the file error without needing this adjustment.